### PR TITLE
ECPINT-1143: handle klarna void action

### PR DESCRIPTION
**Pull request by ved-beejadhur-cko**
**Jira:** [ECPINT-1143](https://checkout.atlassian.net/browse/ECPINT-1143)

### Notes

This Pull Request handles klarna void action on the admin side

### Housekeeping

- [ ] Unit & Integrations tests
- [ ] Updated relevant documentation
- [ ] Manually tested new functionality

### CODE REVIEW

File: src/Helper/Url.php [hunk 1]
```diff
@@ -31,8 +31,16 @@ public function checkPaymentUrl(String $paymentId): string {
     /**
      * cko void payment url
      */
-    public function voidPaymentUrl(String $paymentId) {
-        return self::checkUrlSlash(config::ckoUrl()). 'payments/'. $paymentId . '/voids';
+    public function voidPaymentUrl($param, string $key) {
+        $isLive = self::isLive($key);
+        
+        if($param['payment_method'] == "Klarna") {
```
Comment (mitja-kramberger-cko) [inline src/Helper/Url.php]: This is fine, as Middleware is currently just a pass-through.

File: src/Resources/app/administration/src/components/cko-modal/index.js [hunk 2]
```diff
@@ -1,168 +1,168 @@
...
+    },
+    captureOrder(amountCapture, paymentMethod) {
+      this.CkoCaptureService.capture({
+        payment_id: this.orderInfo.customFields.payment_approved.payment_id,
+        amount: amountCapture,
+        currency: this.orderInfo.currency.isoCode,
+        payment_method: paymentMethod,
+      }).then((response) => {
+        console.log(response);
+        if (
```
Comment (mitja-kramberger-cko) [inline src/Resources/app/administration/src/components/cko-modal/index.js]: Remove `console.log`, and refactor status checks (and page reload) to a common hander.

File: src/Resources/app/administration/src/components/cko-modal/index.js [hunk 3]
```diff
@@ -1,168 +1,168 @@
...
-        fadeCaptureErrorText(){
-            document.getElementById("captureError").innerHTML = "";
-        },
-        fadeRefundErrorText(){
-            document.getElementById("refundError").innerHTML = "";
-        },
-        validate(action,paymentMethod){
-            let inputAmount = document.getElementById("amount").value * 1;
-            let customFields = this.orderInfo.customFields
+      if (paymentMethod === "Sofort" || paymentMethod === "Sepa") {
```
Comment (mitja-kramberger-cko) [inline src/Resources/app/administration/src/components/cko-modal/index.js]: Can we avoid coding against a specific payment method, maybe something like:

```
      if (this.orderInfo.customFields.payment_pending.payment_id) {
        paymentID = this.orderInfo.customFields.payment_pending.payment_id;
      } else {
        paymentID = this.orderInfo.customFields.payment_approved.payment_id;
     }
```

**Review by mitja-kramberger-cko — CHANGES_REQUESTED** [review]
@ved-beejadhur-cko PR comments also apply to https://github.com/checkout/checkout-shopware6-plugin/pull/5

**Comment by ved-beejadhur-cko** [comment]
> @ved-beejadhur-cko PR comments also apply to #5

modification done

**Review by mitja-kramberger-cko — APPROVED** [review]

**Pull request merged**